src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportSession.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            SingleWriter = true
+        });
+
+        private int _disposed;
+        internal Http3WebtransportManager _WebtransportManager;
+
+        internal const string WebTransportProtocolValue = "webtransport";
+        internal const string VersionEnabledIndicator = "1";
+        internal const string SecPrefix = "sec-webtransport-http3-";
+        internal const string VersionHeaderPrefix = $"{SecPrefix}draft";
+        internal const string CurrentSuppportedVersion = $"{VersionHeaderPrefix}02";
+
+        internal Http3WebtransportSession(QuicStream connectStream, Http3WebtransportManager manager)
+        {
+            _WebtransportManager = manager;
+            _connectStream = connectStream;
+            _ = _connectStream.WritesClosed.ContinueWith(async t =>
+            {
+                await DisposeAsync().ConfigureAwait(false);
+
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Current);
+
+        }
+
+        /// <summary>
+        /// Creates a webtransport session by creating a webtransport connect request, sending it to <see cref="Uri">uri</see>.
+        /// </summary>
+        public static async ValueTask<Http3WebtransportSession?> ConnectAsync(Uri uri, HttpMessageInvoker? invoker, CancellationToken cancellationToken)
+        {
+            if (invoker is null)
+            {
+                s_defaultInvoker ??= new HttpMessageInvoker(new HttpClientHandler());
+                invoker = s_defaultInvoker;
+            }
+
+            Http3WebtransportSession? webtransportSession;
+            try
+            {
+                HttpRequestMessage request;
+                request = new HttpRequestMessage(HttpMethod.Connect, uri) { Version = HttpVersion.Version30, VersionPolicy = HttpVersionPolicy.RequestVersionExact };
+                request.Headers.Protocol = WebTransportProtocolValue;
+                Task<HttpResponseMessage> sendTask = invoker is HttpClient client ? client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken) : invoker.SendAsync(request, cancellationToken);
+                HttpResponseMessage response = await sendTask.ConfigureAwait(false);
+                WebtransportHttpContent connectedWebtransSessionContent = (WebtransportHttpContent)response.Content;
+                webtransportSession = connectedWebtransSessionContent.webtransportSession;
+                if (!response.IsSuccessStatusCode || !response.Headers.Contains(Http3WebtransportSession.VersionHeaderPrefix))
+                {
+                    await webtransportSession.AbortIncomingSessionWebtransportStreams((long)Http3ErrorCode.WebtransportBufferedStreamRejected).ConfigureAwait(false);
+                    await webtransportSession.DisposeAsync().ConfigureAwait(false);
+                    throw new(SR.net_webtransport_server_rejected);
```

Comment:
You need to specify the type per code style.
Also, you shouldn't throw base Exception unless it's in tests.